### ISSUE TITLE
handle-done-callback: wrongly detect done callback

### ISSUE THREAD

**Issue by VinhVu0412** [lo1tuma/eslint-plugin-mocha#205]
I am using nightwatch with mocha, my test function look like this
```js
it('uses BDD to run the Google simple test', function(browser) {
  browser
    .url('http://google.com')
    .expect.element('body').to.be.present.before(1000);

  browser.setValue('input[type=text]', ['nightwatch', browser.Keys.ENTER])
    .pause(1000)
    .assert.containsText('#main', 'Night Watch');
});
```
(More detail about using mocha with nightwatch is [here](https://github.com/nightwatchjs/nightwatch-docs/blob/master/guide/running-tests/using-mocha.md))

When I check with eslint, it says 
`error  Expected "browser" callback to be handled  mocha/handle-done-callback`

I am using
```json
{
  "nightwatch": "^1.1.13",
  "mocha": "^6.2.0",
  "eslint": "^6.1.0",
  "eslint-plugin-mocha": "^6.0.0"
}
```

The checker thinks `browser` is `done` callback, but it's not.

**Comment by lo1tuma** [comment]
Mocha always expects the `done` callback parameter to be the first argument. The behavior of injecting the `browser` parameter is added by nightwatch and is not compatible with the default behavior of mocha.

That’s being said, I would be open for a change proposal that solves this kind of issues in a generic way (i.e. not nightwatch specific). Maybe something like adding a configuration option that allows you to specify the expected argument position.

**Comment by VinhVu0412** [comment]
That sounds great. At this point of time, I'll stay with inline ignore.

**Comment by lo1tuma** [comment]
treats the first parameter of a Mocha test callback as the async completion callback by design. For frameworks like Nightwatch that pass a test context object such as  in the first parameter, this rule is not compatible out of the box. The practical resolution is to disable  for those files or use an override for Nightwatch tests.

**Comment by lo1tuma** [comment]
handle-done-callback treats the first parameter of a Mocha test callback as the async completion callback by design. For frameworks like Nightwatch that pass a test context object such as browser in the first parameter, this rule is not compatible out of the box. The practical resolution is to disable mocha/handle-done-callback for those files or use an override for Nightwatch tests.
